### PR TITLE
#820: Correctly place AWS r53 cleanup jobs

### PR DESCRIPTION
**Pull request by achantavy**
Moves route53 cleanup jobs away from aws.sync_one_account() and aws.sync_multiple_accounts() and into the route53 module.

Removes problematic DNS cleanup jobs that are not bound to a specific AWS account.

Updates DNS schema docs and refactors related tests to make them more readable.

### CODE REVIEW

File: cartography/data/jobs/cleanup/aws_dns_cleanup.json [hunk 1]
```diff
@@ -6,53 +6,59 @@
       "iterationsize": 100
     },
     {
-      "query": "MATCH (n:NameServer)-[:NAMESERVER]->(:DNSZone)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+      "query": "MATCH (n:NameServer)-[:NAMESERVER]->(:DNSZone)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n)",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/aws_dns_cleanup.json:9]: `return COUNT(*) as TotalCompleted` is no longer needed as of https://github.com/lyft/cartography/pull/680 where we emit the number of assets affected by a statement.

File: cartography/data/jobs/cleanup/aws_account_dns_cleanup.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-  "statements": [
-  {
-    "query": "MATCH (n:AWSDNSRecord)-[:MEMBER_OF_DNS_ZONE]->(:AWSDNSZone)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}}) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (achantavy) [inline cartography/data/jobs/cleanup/aws_account_dns_cleanup.json:4]: I moved this to aws_dns_cleanup.json.

File: cartography/intel/aws/route53.py [hunk 3]
```diff
@@ -58,14 +58,19 @@ def link_aws_resources(neo4j_session: neo4j.Session, update_tag: int) -> None:
...
     UNWIND {records} as record
-    MERGE (a:DNSRecord:AWSDNSRecord{id: record.id})
-    ON CREATE SET a.firstseen = timestamp(), a.name = record.name, a.type = record.type
-    SET a.lastupdated = {update_tag}, a.value = record.value
-    WITH a,record
-    MATCH (zone:AWSDNSZone{zoneid: record.zoneid})
-    MERGE (a)-[r:MEMBER_OF_DNS_ZONE]->(zone)
-    ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {update_tag}
+        MERGE (a:DNSRecord:AWSDNSRecord{id: record.id})
```
Comment (achantavy) [inline cartography/intel/aws/route53.py:61]: These queries are the same but I've added whitespace to make them easier to read.

File: cartography/data/jobs/cleanup/aws_dns_cleanup.json (hunk 1 above)
Comment (kedarghule) [reply cartography/data/jobs/cleanup/aws_dns_cleanup.json:15]: I had a couple questions.
1) I didn't quite catch why this is being removed. Can you explain? :sweat_smile:
2) Does this need to be done for all cleanup files?

Comment (kedarghule) [reply cartography/data/jobs/cleanup/aws_dns_cleanup.json:15]: Alright!

**Review by ramonpetgrave64 — APPROVED** [review]

**Pull request merged**